src/Modules/UngradedModules/ModuleGens.jl

```diff
@@ -261,22 +261,49 @@ end
 
 Convert a Singular vector to a free module element.
 """
-function (F::FreeMod)(s::Singular.svector)
-  pos = Int[]
-  values = []
+function (F::FreeMod{<:MPolyRingElem})(s::Singular.svector)
   Rx = base_ring(F)
-  R = base_ring(Rx)
-  for (i, e, c) = s
-    f = Base.findfirst(==(i), pos)
-    if f === nothing
-      push!(values, MPolyBuildCtx(base_ring(F)))
-      f = length(values)
-      push!(pos, i)
+  R = coefficient_ring(Rx)
+  ctx = MPolyBuildCtx(Rx)
+  
+  # shortcut in order not to allocate the dictionary
+# if isone(length(s)) # TODO: length doesn't work!
+#   (i, e, c) = first(s)
+#   push_term!(ctx, R(c), e)
+#   return FreeModElem(sparse_row(Qx, [(i, finish(ctx))]))
+# end
```

Comment:
Does anyone know what is the correct way to check whether a `Singular.svector` has only one term? `length` throws an error, even though it should be part of the iterator interface.

Reply:
length(p::svector) = Int(libSingular.pLength(p.ptr))

should do it

Reply:
see https://github.com/oscar-system/Singular.jl/pull/833

Reply:
Update: @hannes14 's PR on Singular.jl just got merged. Do we have to wait for a release there? Or can we remove the method here already? 

ping @lgoettgens

Reply:
You will need to wait for a new release of Singular.jl, and then adapt the `compat` entry in Oscar's `Project.toml` to this new version number